projects/chaos/inc/sequencer.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
```

Comment:
Module comment?

From what I understand, you've basically created a hierarchical state machine where each state acts as a superstate and your arrays of events could be considered as sequential states with transitions back to the initial state.

Reply:
Yes that's the correct interpretation I've added significantly more documentation. Let me know if anything is still unclear.